docs/content/integrations/openai.mdx

```diff
@@ -0,0 +1,144 @@
+---
+title: "Dagster OpenAI"
+description: The dagster-openai library provides the ability to build OpenAI pipelines with Dagster, through helpful resources.
+---
+
+# Dagster OpenAI
+
+This library provides the ability to build OpenAI pipelines with Dagster, through helpful resources.
+
+This library includes a resource, <PyObject module="dagster_openai" object="OpenAIResource" />, using the <a href="https://github.com/openai/openai-python">OpenAI Python API Library</a>, which provides a simple way to interact with OpenAI API.
+
+When used with Dagster's [Software-defined Asset](/concepts/assets/software-defined-assets), the resource automatically logs OpenAI usage metadata in the asset metadata. More details can be found in [relevant APIs](#dagster-openai-relevant-apis).
+
+
+---
+
+## <a name="dagster-openai-relevant-apis"></a>Relevant APIs
+
+| Name                                                                        | Description                                                                           |
+| --------------------------------------------------------------------------- | ------------------------------------------------------------------------------------- |
+| <PyObject module="dagster_openai" object="with_usage_metadata" />           | The function wrapper used on OpenAI API Endpoint methods to log OpenAI usage metadata |
+| <PyObject module="dagster_openai" object="OpenAIResource" />                | The OpenAI Resource used for handing the client                                       |
+
+---
+
+## Getting started
+
+To get started with `dagster-openai` and OpenAI, familiarize yourself with the <a href="https://github.com/openai/openai-python">OpenAI Python API Library</a> by reading their docs which describe how to interact with the OpenAI API using Python.
+
+More information is also available the <a href="https://platform.openai.com/docs/introduction">OpenAI API documentation website</a>.
+
+---
+
+## Prerequisites
+
+To get started, you will need to install the `dagster` and `dagster-openai` Python packages:
+
+```bash
+pip install dagster dagster-openai
+```
+
+Note that you will need an OpenAI [API key](https://platform.openai.com/api-keys) to use the resource, which can be generated in your OpenAI account.
+
+---
+
+## Connecting to OpenAI
+
+The first step in using OpenAI with Dagster is to tell Dagster how to connect to an OpenAI Client using an OpenAI [resource](/concepts/resources). This resource contains the credentials needed to interact with OpenAI API.
+
+We will supply our credentials as environment variables. For more information on setting environment variables in a production setting, see [Using environment variables and secrets](/guides/dagster/using-environment-variables-and-secrets).
+
+```bash
+export OPENAI_API_KEY=...
```

Comment:
lets follow the best practice described in the "Using environment variables and secrets" guide -i.e. make a `.env` file for local use case, instead of bare export